File: esthesis-platform-ui/src/app/dto/infrastructure-report-dto.ts
export interface InfrastructureReportDto {
  clusterLeader: boolean;
  inClusterMode: boolean;
  nodeId: string;
  mqttServerReports: MqttServerReport[];
  zookeeperReport: ZookeeperReport;
}

interface MqttServerReport {
  id: number;
  ipAddress: string;
  leader: boolean;
}

interface ZookeeperReport {
  id: number;
  ipAddress: string;
  leader: boolean;
}

File: esthesis-backend/src/main/java/esthesis/backend/dto/InfrastructureReportDTO.java
package esthesis.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@NoArgsConstructor
@Accessors(chain = true)
public class InfrastructureReportDTO {
  private String nodeId;
  private boolean inClusterMode;
  private boolean clusterLeader;
  private List<MqttServerReport> mqttServerReports;
  private ZookeeperReport zookeeperReport;

  @Data
  @NoArgsConstructor
  @Accessors(chain = true)
  public static class MqttServerReport {
    private long id;
    private String ipAddress;
    private boolean leader;
  }

  @Data
  @NoArgsConstructor
  @Accessors(chain = true)
  public static class ZookeeperReport {
    private long id;
    private String ipAddress;
    private boolean leader;
  }
}
